### PR TITLE
[Fix #13] Add flash messages to project

### LINKED ISSUE #13: Implement flash messages to improve usability
picture

### PR DESCRIPTION
**Pull request by edmilsonefs**

### CODE REVIEW

**Comment by edmilsonefs** [comment]
After your last feedback given by IM I remove the methods from application_controller to simplify and make visible the flash messages on projects_controller.

Note: The integration with devise messages will be worked on another issue: #44 .

---
screenshot_033015_045401_pm
---
screenshot_033015_045516_pm
---
screenshot_033015_045551_pm
---
screenshot_033015_045639_pm

**Comment by hugopl** [comment]
Patch is way better and simpler, but would be nice to not add the partial template call on every thing that need to show flash messages, i.e. put it in the application layout.

Another issue is: I'm not sure what's the best place to put those messages, for now put the flash messages on top of the section title in the application layout, later we would decide a better place.

Also, run rubocop in your patch, it can tell you minor code style issues like the use of double quotes when single quotes is enough.

And last but not least, _to much messages_ :-P, you don't need a to use flash messages everywhere just because now you have them, i.e. remove the flash messages from the `new` actions and use just a single flash message when a error occur in a `update` action.

**Comment by edmilsonefs** [comment]
The changes were done to comply with the requests referred in your last comment.

Let me know if you need something else.

**Comment by hugopl** [comment]
When I created a new project the flash message did show twice here, above and bellow the project title.

File: app/views/projects/show.html.erb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 <h1><%= @project.name %></h1>
 
+<%= render "shared/flash_messages" %>
```
Comment (hugopl) [inline app/views/projects/show.html.erb]: You already show the partial in application layout, so remove this or the flash message will show twice when a new project is created.

File: app/views/shared/_flash_messages.html.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<% flash.each do |type, message| %>
+  <div class="alert alert-<%= flash_message_class_for(type) %>" role="alert"> <%= message %></div>
```
Comment (hugopl) [inline app/views/shared/_flash_messages.html.erb]: You can keep things simple and just do:
```
<div class="alert alert-<%= type %>" role="alert"> <%= message %></div>
```
And avoid the code in the helper.

File: app/assets/stylesheets/common.css.scss [hunk 3]
```diff
@@ -234,3 +234,76 @@ input[type=checkbox]:checked + label:before {
...
+  .alert-link {
+    color: #66512c;
+  }
+}
+
+.alert-danger {
+  background-color: #f2dede;
+  border-color: #ebccd1;
+  color: #a94442;
+  hr {
```
Comment (hugopl) [inline app/assets/stylesheets/common.css.scss]: ditto

File: app/assets/stylesheets/common.css.scss [hunk 4]
```diff
@@ -234,3 +234,76 @@ input[type=checkbox]:checked + label:before {
...
+}
+
+.alert-warning {
+  background-color: #fcf8e3;
+  border-color: #faebcc;
+  color: #8a6d3b;
+  hr {
+    border-top-color: #f7e1b5;
+  }
+  .alert-link {
```
Comment (hugopl) [inline app/assets/stylesheets/common.css.scss]: ditto

File: app/assets/stylesheets/common.css.scss [hunk 5]
```diff
@@ -234,3 +234,76 @@ input[type=checkbox]:checked + label:before {
   content: '*';
   color: red;
 }
+
+/**
+  * Flash messages in a fashion way
+  */
+.alert {
+  padding: 15px;
```
Comment (hugopl) [inline app/assets/stylesheets/common.css.scss]: Too much padding :-), try 4 or 5px

File: app/assets/stylesheets/common.css.scss [hunk 6]
```diff
@@ -234,3 +234,76 @@ input[type=checkbox]:checked + label:before {
...
+  margin-bottom: 20px;
+  border: 1px solid transparent;
```
Comment (hugopl) [inline app/assets/stylesheets/common.css.scss]: Why a transparent border?

File: app/assets/stylesheets/common.css.scss [hunk 7]
```diff
@@ -234,3 +234,76 @@ input[type=checkbox]:checked + label:before {
...
+  border-radius: 4px;
+  h4 {
```
Comment (hugopl) [inline app/assets/stylesheets/common.css.scss]: There's no h4 tag in your alert code, so this CSS has no use.

File: app/assets/stylesheets/common.css.scss [hunk 8]
```diff
@@ -234,3 +234,76 @@ input[type=checkbox]:checked + label:before {
...
+    margin-top: 0;
+    color: inherit;
+  }
+  .alert-link {
```
Comment (hugopl) [inline app/assets/stylesheets/common.css.scss]: Same here, no use for this CSS

File: app/assets/stylesheets/common.css.scss [hunk 9]
```diff
@@ -234,3 +234,76 @@ input[type=checkbox]:checked + label:before {
...
+    font-weight: bold;
+  }
+}
+
+.alert > p,
```
Comment (hugopl) [inline app/assets/stylesheets/common.css.scss]: Same here, no use for this CSS

File: app/assets/stylesheets/common.css.scss [hunk 10]
```diff
@@ -234,3 +234,76 @@ input[type=checkbox]:checked + label:before {
...
+.alert > ul {
```
Comment (hugopl) [inline app/assets/stylesheets/common.css.scss]: Same here, no use for this CSS

File: app/assets/stylesheets/common.css.scss [hunk 11]
```diff
@@ -234,3 +234,76 @@ input[type=checkbox]:checked + label:before {
...
+  margin-bottom: 0;
+}
+.alert > p + p {
```
Comment (hugopl) [inline app/assets/stylesheets/common.css.scss]: Same here, no use for this CSS

File: app/assets/stylesheets/common.css.scss [hunk 12]
```diff
@@ -234,3 +234,76 @@ input[type=checkbox]:checked + label:before {
...
+  margin-top: 5px;
+}
+
+.alert-success {
+  background-color: #dff0d8;
+  border-color: #d6e9c6;
+  color: #3c763d;
+  hr {
```
Comment (hugopl) [inline app/assets/stylesheets/common.css.scss]: Same here, there's no hr tag, so no use for this CSS

File: app/assets/stylesheets/common.css.scss [hunk 13]
```diff
@@ -234,3 +234,76 @@ input[type=checkbox]:checked + label:before {
...
+  .alert-link {
+    color: #2b542c;
+  }
+}
+
+.alert-info {
+  background-color: #d9edf7;
+  border-color: #bce8f1;
+  color: #31708f;
+  hr {
```
Comment (hugopl) [inline app/assets/stylesheets/common.css.scss]: ditto

File: app/assets/stylesheets/common.css.scss [hunk 14]
```diff
@@ -234,3 +234,76 @@ input[type=checkbox]:checked + label:before {
...
+    border-top-color: #a6e1ec;
+  }
+  .alert-link {
```
Comment (hugopl) [inline app/assets/stylesheets/common.css.scss]: Same here, no use for this CSS

File: app/assets/stylesheets/common.css.scss [hunk 15]
```diff
@@ -234,3 +234,76 @@ input[type=checkbox]:checked + label:before {
...
+    color: #245269;
+  }
+}
+
+.alert-warning {
+  background-color: #fcf8e3;
+  border-color: #faebcc;
+  color: #8a6d3b;
+  hr {
```
Comment (hugopl) [inline app/assets/stylesheets/common.css.scss]: ditto

**Comment by hugopl** [comment]
Almost there, btw, could you rebase your code and turn it into a single patch?

File: app/views/projects/show.html.erb (hunk 1 above)
Comment (edmilsonefs) [inline app/views/projects/show.html.erb]: Done locally.

File: app/views/shared/_flash_messages.html.erb (hunk 2 above)
Comment (edmilsonefs) [inline app/views/shared/_flash_messages.html.erb]: Done locally.

File: app/assets/stylesheets/common.css.scss (hunk 3 above)
Comment (edmilsonefs) [inline app/assets/stylesheets/common.css.scss]: Done locally.

File: app/assets/stylesheets/common.css.scss (hunk 4 above)
Comment (edmilsonefs) [inline app/assets/stylesheets/common.css.scss]: Done locally.

File: app/assets/stylesheets/common.css.scss (hunk 7 above)
Comment (edmilsonefs) [inline app/assets/stylesheets/common.css.scss]: Done locally.

File: app/assets/stylesheets/common.css.scss (hunk 8 above)
Comment (edmilsonefs) [inline app/assets/stylesheets/common.css.scss]: Done locally.

File: app/assets/stylesheets/common.css.scss (hunk 9 above)
Comment (edmilsonefs) [inline app/assets/stylesheets/common.css.scss]: Done locally.

File: app/assets/stylesheets/common.css.scss (hunk 10 above)
Comment (edmilsonefs) [inline app/assets/stylesheets/common.css.scss]: Done locally.

File: app/assets/stylesheets/common.css.scss (hunk 11 above)
Comment (edmilsonefs) [inline app/assets/stylesheets/common.css.scss]: Done locally.

File: app/assets/stylesheets/common.css.scss (hunk 12 above)
Comment (edmilsonefs) [inline app/assets/stylesheets/common.css.scss]: Done locally.

File: app/assets/stylesheets/common.css.scss (hunk 13 above)
Comment (edmilsonefs) [inline app/assets/stylesheets/common.css.scss]: Done locally.

File: app/assets/stylesheets/common.css.scss (hunk 14 above)
Comment (edmilsonefs) [inline app/assets/stylesheets/common.css.scss]: Done locally.

File: app/assets/stylesheets/common.css.scss (hunk 15 above)
Comment (edmilsonefs) [inline app/assets/stylesheets/common.css.scss]: Done locally.

File: app/assets/stylesheets/common.css.scss (hunk 5 above)
Comment (edmilsonefs) [inline app/assets/stylesheets/common.css.scss]: Done locally, tried 10px =(. (give me a chance here)

File: app/assets/stylesheets/common.css.scss (hunk 6 above)
Comment (edmilsonefs) [inline app/assets/stylesheets/common.css.scss]: Done locally, removed.

**Comment by edmilsonefs** [comment]
The rebase was done.

**Comment by hugopl** [comment]
Merged in commit 4087d6662ce27aba8bd0ce4fd3e8283e69b0d171.

**Pull request closed without merging**